### DISCUSSION TITLE
How Do I Increase Supabase-JS API Statement Timeout From 3s?

### DISCUSSION THREAD

**Discussion by devwilliamy** [supabase/supabase#27421] [Questions] [answered]
Hi,

I'm using Supabase with my Next.js app and I'm using the supabase-js library to connect to it.
I used an rpc to find out what role and what the timeout is for that role via the API.

This is what I found:
```
[ { user_name: 'anon', statement_timeout: '3s' } ]
```

I've tried updating the statement_timeout but nothing is getting updated.

I've tried:
```
SET ROLE anon;
alter role anon set statement_timeout = '15s';
SELECT current_user;
SHOW statement_timeout;
```

Even after setting the time out and the role and confirming if I do current_user and it says anon, in the web editor it's 2min (which is fine). But even when I try to get it from the API, it still says 3s.

How do I increase that timeout?

**Comment by GaryAustin1** [comment]
https://supabase.com/docs/guides/database/postgres/configuration#timeouts

**Reply by devwilliamy** [reply]
Hi @GaryAustin1 thanks for your response! 

I tried doing that and it didn't work. I still get statement_timeout: '3s'.
Is there something else I need to do to update it?

**Reply by GaryAustin1** [reply, marked as answer]
Seems a step is missing...
```
alter role anon set statement_timeout = '5s';
NOTIFY pgrst, 'reload config';
```
Checked with this postgres function from the API:
```
declare result int;
begin
select setting into result
from pg_settings
where name = 'statement_timeout';
return result;
end;
```

**Reply by scott-wojan** [reply]
Thank you very much.  NOTIFY pgrst, 'reload config'; solved my issue.  Would be great if the docs were updated to include that!

**Reply by GaryAustin1** [reply]
You can generate a PR or an issue in supabase/supabase for that.

**Comment by vkudak** [comment]
Do db.session.refresh(self) after some working time  if you have something in the cycle
